src/main/java/mops/businesslogic/FileQueryBuilder.java

```diff
@@ -0,0 +1,91 @@
+package mops.businesslogic;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+import java.util.List;
+
+@NoArgsConstructor(access = AccessLevel.PACKAGE)
+@SuppressWarnings("PMD.AvoidFieldNameMatchingMethodName")
+public class FileQueryBuilder {
+    /**
+     * List of owner to search for.
+     */
+    private List<String> owners;
+
+    /**
+     * Names of files to search for.
+     */
+    private List<String> fileNames;
+    /**
+     * File types to search for.
+     */
+    private List<String> types;
+    /**
+     * File tags to search for.
+     */
+    private List<String> tags;
```

Comment:
look at the FileInfo builder to see how collections are built

Reply:
you don't set them, you add to existing ones

Reply:
ok I will try that